src/core/net/srp_server.cpp

```diff
@@ -1760,12 +1760,22 @@ TimeMilli Server::Service::GetKeyExpireTime(void) const
 void Server::Service::GetLeaseInfo(LeaseInfo &aLeaseInfo) const
 {
     TimeMilli now           = TimerMilli::GetNow();
-    TimeMilli expireTime    = GetExpireTime();
     TimeMilli keyExpireTime = GetKeyExpireTime();
 
-    aLeaseInfo.mLease             = Time::SecToMsec(GetLease());
+    if (!mIsDeleted)
+    {
+        TimeMilli expireTime = GetExpireTime();
+
+        aLeaseInfo.mLease          = Time::SecToMsec(GetLease());
+        aLeaseInfo.mRemainingLease = (now <= expireTime) ? (expireTime - now) : 0;
+    }
+    else
+    {
+        aLeaseInfo.mLease          = 0;
```

Comment:
I think it's useful to keep the `mLease` as it is so that the user knows the value granted for this Host/Service (even after it's deleted).

Reply:
Good point. Thanks Changed it in the new push. 

I agree it can be helpful for `Service`. For `Host` it may not matter since we set its `mLease` to zero when we delete it.